### PR TITLE
Protocol modelisation and json serialization

### PR DESCRIPTION
**Pull request by GabrielFleischer**
Create models for the protocol and add a json serialization using Gson

### CODE REVIEW

File: app/src/main/java/com/github/dedis/student20_pop/model/network/level/package-info.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Classes used to model the levels of the protocol
+ **/
+package com.github.dedis.student20_pop.model.network.level;
```
Comment (celinecamacho) [inline app/src/main/java/com/github/dedis/student20_pop/model/network/level/package-info.java:4]: Could you also do a package-info for level.low, level.mid and level.high?

**Review by celinecamacho — APPROVED** [review]
Nice, maybe add a package-info for every level too (low, mid and high)

File: app/src/main/java/com/github/dedis/student20_pop/model/network/level/package-info.java (hunk 1 above)
Comment (GabrielFleischer) [reply app/src/main/java/com/github/dedis/student20_pop/model/network/level/package-info.java:4]: Done

**Review by AlexisTabin — APPROVED** [review]
Great job 👏🏻

**Review by jbmichel14 — APPROVED** [review]

**Pull request merged**